appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/YaFormEditor.java

```diff
@@ -939,6 +945,88 @@ public void onResetDatabase() {
     }
   }
 
+
+  private void registerKeyDownListeners () {
+    RootPanel.get().addDomHandler(new KeyDownHandler() {
```

Comment:
It would be nice to add some comments here summarizing what you're doing with the KeyDownHandler, e.g. what keystrokes you're checking for on what parts of the UI. It's easy for that to get lost in the wall of code.